### PR TITLE
[DDL] More refactor around `AlterTableArgs`

### PR DESCRIPTION
**Pull request by Tang8330**
1. Creating tables is all going through `shared.CreateTable`
2. Removed the create table functionality from `AlterTableArgs`

### CODE REVIEW

File: clients/shared/table.go [hunk 1]
```diff
@@ -25,6 +25,6 @@ func CreateTable(ctx context.Context, dwh destination.DataWarehouse, tableData *
 	}
 
 	// Update cache with the new columns that we've added.
-	tc.MutateInMemoryColumns(true, constants.Add, tableData.ReadOnlyInMemoryCols().GetColumns()...)
+	tc.MutateInMemoryColumns(false, constants.Add, tableData.ReadOnlyInMemoryCols().GetColumns()...)
```
Comment (Tang8330) [inline clients/shared/table.go:28]: This should have been false to begin with. I'm going to put up another PR to remove this flag.

File: lib/destination/ddl/ddl.go [hunk 2]
```diff
@@ -75,7 +75,6 @@ type AlterTableArgs struct {
 	// ContainsOtherOperations - this is sourced from tableData `containOtherOperations`
 	ContainOtherOperations bool
 	TableID                sql.TableIdentifier
-	CreateTable            bool
```
Comment (Tang8330) [inline lib/destination/ddl/ddl.go:78]: I removed the ability to create table in this function

File: clients/shared/table.go (hunk 1 above)
Comment (Tang8330) [reply clients/shared/table.go:28]: https://github.com/artie-labs/transfer/pull/1041

**Review by danafallon — APPROVED** [review]
